File: frontend/src/types/notification.ts
export type NotificationType = 1 | 2 | 3; // 1=BudgetAlert, 2=SalaryReceived, 3=WeeklySummary

export interface Notification {
  id: string;
  type: NotificationType;
  title: string;
  message: string;
  isRead: boolean;
  createdAt: string;
}

File: backend/src/ExpenseTracker.Domain/Entities/Notification.cs
using ExpenseTracker.Domain.Enums;

namespace ExpenseTracker.Domain.Entities;

public class Notification : IUserOwnedEntity
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public User? User { get; set; }
    public NotificationType Type { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Message { get; set; } = string.Empty;
    public bool IsRead { get; set; } = false;
    public string? DeduplicationKey { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
